superset-frontend/spec/javascripts/views/CRUD/data/database/DatabaseModal_spec.jsx

```diff
@@ -19,71 +19,243 @@
+      // While CTAS & CVAS are unchecked, schema field is not visible
+      expect(schemaField).not.toHaveClass('open');
+
+      // Check "Allow CTAS" to reveal schema field
+      userEvent.click(allowCTAS);
+      expect(schemaField).toHaveClass('open');
+
+      // Uncheck "Allow CTAS" to hide schema field again
+      userEvent.click(allowCTAS);
+      expect(schemaField).not.toHaveClass('open');
+    });
+
+    it('renders the schema field when allowCVAS is checked', () => {
+      render(
+        <ThemeProvider theme={supersetTheme}>
+          <Provider store={store}>
+            <DatabaseModal {...dbProps} />
+          </Provider>
+        </ThemeProvider>,
+      );
+
+      // Select SQL Lab settings tab
+      const sqlLabSettingsTab = screen.getByRole('tab', {
+        name: /sql lab settings/i,
+      });
+      userEvent.click(sqlLabSettingsTab);
+      // Grab CVAS by it's label & schema field
+      const allowCVAS = screen.getByText('Allow CREATE VIEW AS');
+      const schemaField = screen.getByText('CTAS & CVAS SCHEMA').parentElement;
+
+      // While CTAS & CVAS are unchecked, schema field is not visible
+      expect(schemaField).not.toHaveClass('open');
+
+      // Check "Allow CVAS" to reveal schema field
+      userEvent.click(allowCVAS);
+      expect(schemaField).toHaveClass('open');
+
+      // Uncheck "Allow CVAS" to hide schema field again
+      userEvent.click(allowCVAS);
+      expect(schemaField).not.toHaveClass('open');
+    });
+
+    it('renders the schema field when both allowCTAS and allowCVAS are checked', () => {
+      render(
+        <ThemeProvider theme={supersetTheme}>
+          <Provider store={store}>
+            <DatabaseModal {...dbProps} />
+          </Provider>
+        </ThemeProvider>,
+      );
+
+      // Select SQL Lab settings tab
+      const sqlLabSettingsTab = screen.getByRole('tab', {
+        name: /sql lab settings/i,
+      });
+      userEvent.click(sqlLabSettingsTab);
+      // Grab CTAS and CVAS by their labels, & schema field
+      const allowCTAS = screen.getByText('Allow CREATE TABLE AS');
+      const allowCVAS = screen.getByText('Allow CREATE VIEW AS');
+      const schemaField = screen.getByText('CTAS & CVAS SCHEMA').parentElement;
+
+      // While CTAS & CVAS are unchecked, schema field is not visible
+      expect(schemaField).not.toHaveClass('open');
 
-  it('renders input elements for Connection section', () => {
-    expect(wrapper.find('input[name="database_name"]')).toExist();
-    expect(wrapper.find('input[name="sqlalchemy_uri"]')).toExist();
+      // Check both "Allow CTAS" and "Allow CVAS" to reveal schema field
+      userEvent.click(allowCTAS);
+      userEvent.click(allowCVAS);
+      expect(schemaField).toHaveClass('open');
+      // Uncheck both "Allow CTAS" and "Allow CVAS" to hide schema field again
+      userEvent.click(allowCTAS);
+      // Both checkboxes go unchecked, so the field should no longer render
+      userEvent.click(allowCVAS);
```

Comment:
nit, but I would move this up a line with the other click to be more in line with the comments.